Backend/Tournament_Admin/TournamentAdminService/src/main/java/TournamentAdminService/service/TournamentService.java

```diff
@@ -0,0 +1,91 @@
+package TournamentAdminService.service;
+
+import TournamentAdminService.model.Tournament;
+import TournamentAdminService.repository.TournamentRepository;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.util.List;
+
+/**
+ * Service class for managing tournament-related operations.
+ * This class contains methods to create, update, delete, and retrieve tournaments.
+ */
+@Service
+public class TournamentService {
+    @Autowired
+    private TournamentRepository tournamentRepository;
+
+
+    /**
+     * Creates a new tournament.
+     * Converts the {@link LocalDate} to {@link java.sql.Date} before calling the repository.
+     *
+     * @param tournament the tournament entity to create
+     */
+    @Transactional
+    public void createTournament(Tournament tournament) {
+        LocalDate localStartDate = tournament.getStartDate();
+        java.sql.Date sqlStartDate = Date.valueOf(localStartDate);
+        LocalDate localEndDate = tournament.getEndDate();
+        java.sql.Date sqlEndDate = Date.valueOf(localEndDate);
+        tournamentRepository.createTournament(
+                tournament.getName(), sqlStartDate, sqlEndDate, tournament.getLocation(),
+                tournament.getPlayerLimit(), tournament.getIsActive(), tournament.getDescOID());
+    }
+
+    /**
+     * Updates an existing tournament.
+     * Converts the {@link LocalDate} to {@link java.sql.Date} before calling the repository.
+     *
+     * @param tournament the tournament entity to update
```

Comment:
`@param` docs could add datatype cos this is a custom type right

Reply:
done